build_tools/circle/build_test_pypy.sh

```diff
@@ -2,31 +2,28 @@
 set -x
 set -e
 
+# System build tools
 apt-get -yq update
-apt-get -yq install libatlas-base-dev liblapack-dev gfortran ccache libopenblas-dev
+apt-get -yq install wget bzip2 build-essential ccache
 
-pip install virtualenv
+# Install pypy and all the scikit-learn dependencies from conda-forge. In
+# particular, we want to install pypy compatible binary packages for numpy and
+# scipy as it would be to costly to build those from source.
+conda install -y mamba
+mamba create -n pypy -y \
+    pypy numpy scipy cython \
+    joblib threadpoolctl pillow pytest \
+    sphinx numpydoc docutils
 
-if command -v pypy3; then
-    virtualenv -p $(command -v pypy3) pypy-env
-elif command -v pypy; then
-    virtualenv -p $(command -v pypy) pypy-env
-fi
-
-source pypy-env/bin/activate
+eval "$(conda shell.bash hook)"
```

Comment:
Setting conda require quite a few steps..

Reply:
Yeah, that was really hell. circle ci must also be doing something weird when passing the bash commands to execute in the container because some setups that would work in my local docker would not work on circle ci. And I could never get micromamba setup to work at all on circle ci (while it would work well and fast in my local setup).